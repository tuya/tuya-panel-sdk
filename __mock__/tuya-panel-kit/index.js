import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
