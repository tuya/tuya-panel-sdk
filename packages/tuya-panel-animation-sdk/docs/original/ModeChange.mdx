## 简介

_负责人: Ivvan（黄婷婷）

`ModeChange` 是 模式切换组件，直观的模式更改体验。

## 代码演示

### 基础使用

<Playground>
  {() => {
    const [state, set] = React.useState({ imgIndex: 0 });
    const setState = value => set({ ...state, ...value });
    const imgUrlList = [
      'https://images.tuyacn.com/rms-static/3f5ccbb0-94bc-11ea-8e3b-e3ecba4013ec-1589334682603.png?tyName=speed1.png',
      'https://images.tuyacn.com/rms-static/3f750ea0-94bc-11ea-a593-57b2fa093611-1589334682762.png?tyName=speed2.png',
      'https://images.tuyacn.com/rms-static/3f773180-94bc-11ea-8e3b-e3ecba4013ec-1589334682776.png?tyName=speed3.png',
    ];
    return (
      <View style={{ height:'100%', alignItems: 'center', justifyContent: 'center' }}>
        <TouchableOpacity
          activeOpacity={1}
          onPress={() => {
            setState({ imgIndex: state.imgIndex === imgUrlList.length - 1  ? 0 : state.imgIndex + 1 });
          }}
        >
          <ModeChange.Scale imgUrl={imgUrlList[state.imgIndex]}  imgStyle={{ tintColor: 'red',width:30,height:30 }}/>
        </TouchableOpacity>
        <ModeChange.Move
          imgUrl={imgUrlList[state.imgIndex]}
          imgStyle={{ tintColor: 'red',width:80,height:80 }}
          onStartAnimted={() => {
            console.log('start');
          }}
          onEndAnimted={() => {
            console.log('end');
          }}
        />
      </View>
     );
  }}
</Playground>