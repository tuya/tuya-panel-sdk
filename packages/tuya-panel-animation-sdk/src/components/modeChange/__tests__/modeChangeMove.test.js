import React from 'react';
import { shallow } from 'enzyme';
import { Animated } from 'react-native';
import { IconFont } from 'tuya-panel-kit';
import ModeChange from '../index';

const imgUrlList =
  'https://images.tuyacn.com/rms-static/3f5ccbb0-94bc-11ea-8e3b-e3ecba4013ec-1589334682603.png?tyName=speed1.png';

const nextUrlList =
  'https://images.tuyacn.com/rms-static/3f750ea0-94bc-11ea-a593-57b2fa093611-1589334682762.png?tyName=speed2.png';

describe('ModeChange.Scale components', () => {
  it('basic render', () => {
    let imgUrl = imgUrlList;
    const wrapper = shallow(
      <ModeChange.Move
        imgUrl={imgUrl}
        imgStyle={{ tintColor: 'red', width: 80, height: 80 }}
        onStartAnimted={jest.fn()}
        onEndAnimted={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
    imgUrl = nextUrlList;
    wrapper.setProps({ imgUrl });
    Animated.parallel = jest.fn((height, config) => {
      return {
        start: jest.fn(callback => callback()),
      };
    });
  });

  it('renderContent render', () => {
    let imgUrl = imgUrlList;
    const wrapper = shallow(
      <ModeChange.Move
        imgUrl={imgUrl}
        imgStyle={{ tintColor: 'red', width: 80, height: 80 }}
        renderContent={<IconFont name="power" />}
        onStartAnimted={jest.fn()}
        onEndAnimted={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
