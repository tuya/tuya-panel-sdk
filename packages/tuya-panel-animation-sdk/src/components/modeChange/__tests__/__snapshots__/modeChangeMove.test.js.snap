// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeChange.Scale components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 160,
        "justifyContent": "center",
        "width": 160,
      },
      null,
      false,
    ]
  }
>
  <AnimatedComponent
    source={
      Object {
        "uri": "https://images.tuyacn.com/rms-static/3f5ccbb0-94bc-11ea-8e3b-e3ecba4013ec-1589334682603.png?tyName=speed1.png",
      }
    }
    style={
      Array [
        Object {
          "height": 80,
          "resizeMode": "stretch",
          "tintColor": "#fff",
          "width": 80,
        },
        Object {
          "height": 80,
          "tintColor": "red",
          "width": 80,
        },
        Object {
          "opacity": 0,
        },
        Object {
          "transform": Array [
            Object {
              "translateY": 20,
            },
          ],
        },
      ]
    }
  />
</View>
`;

exports[`ModeChange.Scale components renderContent render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 160,
        "justifyContent": "center",
        "width": 160,
      },
      null,
      false,
    ]
  }
>
  <Component
    name="power"
  />
</View>
`;
